client-app/shared/cart/components/add-to-cart.vue

```diff
@@ -190,11 +187,27 @@ function onKeypress(event: KeyboardEvent) {
 /**
  * Limit max value.
  */
-function onInput() {
+function onInput(): void {
   if (!enteredQuantity.value) {
     enteredQuantity.value = undefined;
   } else if (enteredQuantity.value > MAX_VALUE) {
     enteredQuantity.value = MAX_VALUE;
+  }
+
+  if (!enteredQuantity.value) {
+    return;
+  }
+
+  if (
+    props.product.minQuantity &&
+    props.product.maxQuantity &&
+    (enteredQuantity.value < props.product.minQuantity || enteredQuantity.value > props.product.maxQuantity)
+  ) {
+    setErrors(t("shared.cart.add_to_cart.errors.min_max", [props.product.minQuantity, props.product.maxQuantity]));
+  } else if (props.product.minQuantity && enteredQuantity.value < props.product.minQuantity) {
+    setErrors(t("shared.cart.add_to_cart.errors.min", [props.product.minQuantity]));
+  } else if (props.product.maxQuantity && enteredQuantity.value > props.product.maxQuantity) {
+    setErrors(t("shared.cart.add_to_cart.errors.max", [props.product.maxQuantity]));
```

Comment:
Replace using this:
https://github.com/jquense/yup?tab=readme-ov-file#schemawithmutationbuilder-current-schema--void-void and
https://github.com/jquense/yup?tab=readme-ov-file#schematestname-string-message-string--function--any-test-function-schema or
https://github.com/jquense/yup?tab=readme-ov-file#schematestoptions-object-schema